dataservice/util/data_gen/data_generator.py

```diff
@@ -208,11 +217,12 @@ def _create_participants(self, total):
         creates participants with samples, demographics, and diagnoses
         """
         for i in range(total):
+            diagnoses = self._create_diagnoses(
+                random.randint(self.min_samples, self.max_diagnoses))
+            self._get_unique_sites(diagnoses)
```

Comment:
It's really difficult to know what this is doing without going back and reading through the entire code. Using global state to store intermediate results leads to a lot of headaches and counter-intuitive behavior. Looking at this scope, I cannot tell what depends on the value being calculated here and I would (wrongly) think that I could reposition this line anywhere ins the function.

Why not determine the unique sites, then return them, and pass them into the functions that need them explicitly?

Also, you could probably get away with doing this in one line so the reader doesn't have to trace through another function:
```
unique_sites = { d.tumor_location for d in diagnoses }
```